### ISSUE TITLE
Middleware in Django >= 1.10

### ISSUE THREAD

**Issue by justinrporter** [pinax/pinax-stripe#347]
Hi all,

I hope this hasn't already been addressed--I searched around on the issue tracker and tried with with b5b05d2 as well as 3.4.1 from PyPI.

When I use the middleware, `pinax.stripe.middleware.ActiveSubscriptionMiddleware`, I get:

```
Unhandled exception in thread started by <function wrapper at 0x10ca01b90>
Traceback (most recent call last):
  File "/Users/jrporter/projects/.venvs/studentpt/lib/python2.7/site-packages/django/utils/autoreload.py", line 227, in wrapper
    fn(*args, **kwargs)
  File "/Users/jrporter/projects/.venvs/studentpt/lib/python2.7/site-packages/django/core/management/commands/runserver.py", line 147, in inner_run
    handler = self.get_handler(*args, **options)
  File "/Users/jrporter/projects/.venvs/studentpt/lib/python2.7/site-packages/django/contrib/staticfiles/management/commands/runserver.py", line 28, in get_handler
    handler = super(Command, self).get_handler(*args, **options)
  File "/Users/jrporter/projects/.venvs/studentpt/lib/python2.7/site-packages/django/core/management/commands/runserver.py", line 68, in get_handler
    return get_internal_wsgi_application()
  File "/Users/jrporter/projects/.venvs/studentpt/lib/python2.7/site-packages/django/core/servers/basehttp.py", line 47, in get_internal_wsgi_application
    return import_string(app_path)
  File "/Users/jrporter/projects/.venvs/studentpt/lib/python2.7/site-packages/django/utils/module_loading.py", line 20, in import_string
    module = import_module(module_path)
  File "/usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/Users/jrporter/projects/studentpt/studentpt/studentpt/wsgi.py", line 18, in <module>
    application = get_wsgi_application()
  File "/Users/jrporter/projects/.venvs/studentpt/lib/python2.7/site-packages/django/core/wsgi.py", line 14, in get_wsgi_application
    return WSGIHandler()
  File "/Users/jrporter/projects/.venvs/studentpt/lib/python2.7/site-packages/django/core/handlers/wsgi.py", line 151, in __init__
    self.load_middleware()
  File "/Users/jrporter/projects/.venvs/studentpt/lib/python2.7/site-packages/django/core/handlers/base.py", line 82, in load_middleware
    mw_instance = middleware(handler)
TypeError: object() takes no parameters
```

Which seems to be a Django >= 1.10 middleware compatibility issue, as outlined by [this StackOverflow post](https://stackoverflow.com/questions/42232606/django-exception-middleware-typeerror-object-takes-no-parameters).

The thing that's confusing me is that I noticed that the current builds actually pass on Django 1.10 and 1.11... maybe I missed somthing? Or the middleware isn't part of the tests?

**Comment by askz** [comment]
Hi @justinrporter I can confirm having the same issue with django 11....
did you managed to get a workaround or something for that ?

**Comment by gregnewman** [comment]
@askz @justinrporter I'm trying to reproduce this under Django 1.11.4 and cannot.  

1. Can you tell me what your middleware settings look like?  
2. Are you using a customer with an active subscription?

**Comment by justinrporter** [comment]
Sorry, I should have actually provided steps to reproduce:

1. Start a new project with an empty app and Django >= 1.10. I just tested it with 1.11.4 and it's still there.
2. To `settings.py`'s `INSTALLED_APPS` add the empty app, `pinax.stripe`.
3. To `MIDDLEWARE`, add `pinax.stripe.middleware.ActiveSubscriptionMiddleware`
4. (Add `SITE_ID = 1` to `settings.py`)
5. Run `./manage.py runserver`

The django-stripe actually had the same problem, which was fixed in dj-stripe/dj-stripe#543.

The issue is that the way middleware is processed in Django >= 1.10 is different. The fix is trivial, though--just add the base class of `ActiveSubscriptionMiddleware` from `object` to `MiddlewareMixin`.
